cabal-install/src/Distribution/Client/Check.hs

```diff
@@ -56,22 +56,9 @@ check verbosity = do
     (ws, ppd) <- readGenericPackageDescriptionCheck verbosity pdfile
     -- convert parse warnings into PackageChecks
     let ws' = map (wrapParseWarning pdfile) ws
-    -- flatten the generic package description into a regular package
-    -- description
-    -- TODO: this may give more warnings than it should give;
-    --       consider two branches of a condition, one saying
-    --          ghc-options: -Wall
-    --       and the other
-    --          ghc-options: -Werror
-    --      joined into
-    --          ghc-options: -Wall -Werror
-    --      checkPackages will yield a warning on the last line, but it
-    --      would not on each individual branch.
-    --      However, this is the same way hackage does it, so we will yield
-    --      the exact same errors as it will.
     let pkg_desc = flattenPackageDescription ppd
     ioChecks <- checkPackageFiles verbosity pkg_desc "."
-    let packageChecks = ioChecks ++ checkPackage ppd (Just pkg_desc) ++ ws'
+    let packageChecks = ioChecks ++ checkPackage ppd ++ ws'
```

Comment:
It’s a pity that flattened `pkg_desc` is still needed — for IO checks. Is there no way around it?

Reply:
There is a way around it. I have passed two months rewriting this and I forgot to change functions, how embarassing!

Note that `checkPackageFiles` cannot be deprecated as now because there are still some part of the codebase requiring a check on PD rather than GPD (e.g. `SrcDist.hs`).